src/languageTools/DefaultProviders.js

```diff
@@ -371,6 +376,23 @@ define(function (require, exports, module) {
         this._results.set(filePath, {
             errors: errors
         });
+        if(this._defferedLintingPromise.get(filePath)) {
+           this._defferedLintingPromise.get(filePath).resolve(this._results.get(filePath));
+           this._defferedLintingPromise.delete(filePath);
+        }
+        if (this._ValidateOnType) {
```

Comment:
Few issues: 

- Check if filePath === currentActiveDocumentPath? No need to run inspection for files which are not active.
- How is onSave being handled here?
- Why do we need a promiseMap here at all? The request is being run once we have already gotten the result.

Reply:
@shubhsnov  on File Save and File Open diagnostics would be run by CodeInspection Manager itself.
in case of file open Diagnostics result won't be available immediately in that case result will be return using promise

Reply:
Okay. If the promise is only for the file open scenario then why do we require a Map? We can just keep a promise for the current doc since we are already not running the code inspection request for any inactive docs.

Reply:
FileOpen scenario is the the sceanrio which I could think where result might not be available. there could be other scenarios as well.

Reply:
In any case, we only show the linting for the current file. Only having currentActiveDocPromise should suffice. No point in caching since the cached result will never be used.

Reply:
Also please add a check in the getInspectionResultsAsync for filePath === currentActiveDocumentPath as well.

Reply:
> In any case, we only show the linting for the current file. Only having currentActiveDocPromise should suffice. No point in caching since the cached result will never be used.

using just variable will cause problem. If you will see the code workflow It will not cache result if it is not to be used.  To me it looks perfectly ok. we can discuss this if you like.

Reply:
> Also please add a check in the getInspectionResultsAsync for filePath === currentActiveDocumentPath as well.

already fixed

Reply:
> > Also please add a check in the getInspectionResultsAsync for filePath === currentActiveDocumentPath as well.
> 
> already fixed

I specifically meant for `getInspectionResultsAsync` not `setInspectionResultsAsync`. It would act as a guard and optimization. Suppose a file has hundred of errors, but we switched to any other file before actually getting the results. Why would we want those hundred of results to pass through another function callback and then get discarded, since the active file changed. Doesn't seem proper to me. If we have a check in set, we should have a check in get.

Reply:
Not Sure if you are looking at complete code workflow from CodeInspection side. It looks perfectly ok to me. the guard you are mentioning is already in place in codeInspection. CodeInspection always call getInspectionResultsAsync  for current File only.
and this function can't return null, It has to return promise.

If still not clear we can discuss this.

Reply:
It's an optimization.  Okay let me explain it to you more clearly here:
You open File A (Some large file which let's assume will cause some time for the diagnostics response) 
getInspectionResultsAsync will get called
Just as the function was called you switched the document
Now you are inside the getInspectionResultsAsync for File A but the active Document is File B

Now according to me, instead of creating a promise and waiting for the promise to resolve, we can instead do $.Deferred().resolve(null) because it won't matter, as CodeInspection will drop the promise as it is.

Reply:
not taking this change as I am not convinced that it is an optimization. we are just double checking if the file is current file.